include/klee/Internal/System/Time.h

```diff
@@ -10,20 +10,112 @@
 #ifndef KLEE_UTIL_TIME_H
 #define KLEE_UTIL_TIME_H
 
-#include "llvm/Support/TimeValue.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <chrono>
+#include <string>
+#include <tuple>
+#include <sys/time.h>
 
 namespace klee {
-  namespace util {
+  namespace time {
+
+    /// The klee::time namespace offers various functions to measure the time (`getWallTime`)
+    /// and to get timing information for the current KLEE process (`getUserTime`).
+    /// This implementation is based on `std::chrono` and uses time points and time spans.
+    /// For KLEE statistics, spans are converted to µs and stored in `uint64_t`.
+
+    using SteadyTimePoint = std::chrono::steady_clock::time_point;
+    using Duration = std::chrono::steady_clock::duration;
+
+    struct Point;
+    struct Span;
+
+    /// Returns information about clock
+    std::string getClockInfo();
+
+    /// Returns pair (user time, system time) of durations
+    std::pair<Span, Span> getElapsedTime();
+
+    /// Returns point in time using a monotonic steady clock
+    Point getWallTime();
+
+    /// Parses a string to span
+    Span spanFromString(const std::string&);
```

Comment:
Why is this not just a `Span` constructor?

Reply:
Because it doesn't need access to members:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-member

Reply:
But you already have a function that is a member (the constructor), so I don't think this applies.

Reply:
I see, makes sense.